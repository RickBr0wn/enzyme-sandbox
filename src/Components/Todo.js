import React from 'react'
import TodoItem from './TodoItem'
import uuid from 'react-uuid'

const intialValue = {
  todos: ['Go jogging', 'Ride a bike', 'Write more tests']
}

const reducer = (state, action) => {
  switch (action.type) {
    default:
      return state
  }
}

const Todo = () => {
  const [state] = React.useReducer(reducer, intialValue)
  const { todos } = state
  return (
    <div className='todo-container'>
      <h1>TODO Component</h1>
      <ul>
        {todos.map(_todo => (
          <TodoItem todo={_todo} key={uuid()} />
        ))}
      </ul>
    </div>
  )
}

export default Todo
