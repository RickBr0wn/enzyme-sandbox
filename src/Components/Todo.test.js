import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import Todo from './Todo'

configure({ adapter: new Adapter() })

describe('<Todo />', () => {
  it('should render correctly', () => {
    const todoComponent = renderer.create(<Todo />)
    const tree = todoComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should display a list of todo items', () => {
    const todoComponent = shallow(<Todo />)
    const li = todoComponent.find('li')
  })
})
