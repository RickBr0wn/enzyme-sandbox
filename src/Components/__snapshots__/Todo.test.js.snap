// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Todo /> should render correctly 1`] = `
<div
  className="todo-container"
>
  <h1>
    TODO Component
  </h1>
  <ul>
    <li
      className="todo-item"
    >
      Go jogging
    </li>
    <li
      className="todo-item"
    >
      Ride a bike
    </li>
    <li
      className="todo-item"
    >
      Write more tests
    </li>
  </ul>
</div>
`;
